packages/atomic/src/components/search/atomic-search-box/atomic-search-box.tsx

```diff
@@ -153,11 +153,18 @@ export class AtomicSearchBox {
   @Prop() public suggestionTimeout = 400;
 
   /**
-   * Whether to prevent the user from triggering a search from the component.
-   * Perfect for use cases where you need to disable the search conditionally, like when the input is empty.
+   * Prevent the user from triggering a search from the component.
+   * To be used when you need to disable the search conditionally.
+   * For the case when you need to disable the search on no query input refer to {@link disableSearchWhenNoQuery}.
    */
   @Prop({reflect: true}) public disableSearch = false;
 
+  /**
+   * Disable search button when there is no query input.
+   */
+  // TODO(major release): Consider combining with `disableSearch` prop and/or defaulting to true.
+  @Prop({reflect: true}) public disableSearchWhenNoQuery = false;
```

Comment:
An alternative, which could be more flexible, would be to change the option to something akin to "minimumInputContent", which would disable search when there is less than X characters in the search box.

This is a semi frequent request from implementers, that is being able to disable search when there is less than 2, or 3 characters. 

0 character is the most frequent need, but I think we would be able to cover a bit more ground with an option that would accept a number instead.

Reply:
I'm starting to think we should ask UX about this.

I can kind of imagine some purpose for disabling the search box while it's empty, e.g.:
* In case somebody accidentally presses enter.
* Because it just doesn't make much logical sense to have search results without a query.

However, I can't really think of a purpose for disabling search under a certain number of characters, and I'm not sure if that's something we even want to allow since I can only imagine it being an annoyance.

Concerning the naming of this parameter, I think we could still make it a boolean, but instead make it positive (`enable` instead of `disable`).

For example: `@Prop({reflect:true}) public allowEmptyQueries = true`

Reply:
> However, I can't really think of a purpose for disabling search under a certain number of characters, and I'm not sure if that's something we even want to allow since I can only imagine it being an annoyance.

Often time, the reason is $  😉 .

Some client (for good and/or bad reasons) want to limit their consumption against the Coveo platform, because that's what contracts are based on. And so they go on some crusade to find everything they can configure/tweak to achieve that goal. And trying to block/prevent "incomplete" search requests is something we see from time to time.

As for if there's good UX reasons behind it, I don't think so.

Reply:
What do you think about having this prop control triggering search suggestions based on query input length instead of the search action itself. That way if the user wanted to search for small words less than the set length, they can still do so (without being confused/annoyed about why the search is not enabled) but it would also satisfy the goal of limiting resource consumption?

Reply:
yeah, we can add the option to stop suggestions later

Reply:
https://coveo.slack.com/archives/G7HD9P48H/p1681832351521629